src/internal_helper.cc

```diff
@@ -356,6 +356,55 @@ Status BackupRestore::backup(FileOps* f_ops,
    }
 }
 
+Status BackupRestore::backup(FileOps* f_ops,
+                             FileHandle** file,
+                             const std::string& filename,
+                             const SizedBuf& ctx,
+                             size_t length,
+                             size_t bytes_to_skip,
+                             bool call_fsync)
+{
+    Status s;
+    std::string dst_file = filename + ".bak";
+
+    try {
+        if (!f_ops->exist(dst_file)) {
```

Comment:
Shall we need to check whether the file exist in each call? I tested it and even if the file is removed, all system call will success, but no file is persisted.

Reply:
Do you mean if `dst_file` does not exist, but `*file != nullptr`, all below `pwrite` succeeds, but nothing is written? Then yes.

Reply:
Yes. So, shall we need to rebuild the *file pointer here? Or we can ignore this case?

Reply:
Yes we should, otherwise backup won't be working.